shell/common/rasterizer.cc

```diff
@@ -393,42 +421,31 @@ fml::Milliseconds Rasterizer::GetFrameBudget() const {
 
 Rasterizer::DoDrawResult Rasterizer::DoDraw(
     std::unique_ptr<FrameTimingsRecorder> frame_timings_recorder,
-    std::unique_ptr<flutter::LayerTree> layer_tree,
-    float device_pixel_ratio) {
+    std::list<LayerTreeTask> tasks) {
   TRACE_EVENT_WITH_FRAME_NUMBER(frame_timings_recorder, "flutter",
                                 "Rasterizer::DoDraw", /*flow_id_count=*/0,
                                 /*flow_ids=*/nullptr);
   FML_DCHECK(delegate_.GetTaskRunners()
                  .GetRasterTaskRunner()
                  ->RunsTasksOnCurrentThread());
+  frame_timings_recorder->AssertInState(FrameTimingsRecorder::State::kBuildEnd);
```

Comment:
Should this check be before `TRACE_EVENT_WITH_FRAME_NUMBER` as that mutates the `frame_timings_recorder`?

Reply:
It seems to me that the `TRACE` only gets `recorder->GetFrameNumberTraceArg()` and does not mutate the recorder at all.